lib/puppet/pops/loaders.rb

```diff
@@ -253,7 +253,8 @@ def private_loader_for_module(module_name)
 
   def add_loader_by_name(loader)
     name = loader.loader_name
-    if @loaders_by_name.include?(name)
+    # If a Bolt project is defined
+    if @loaders_by_name.include?(name) && @project.name != name
```

Comment:
```suggestion
    if @loaders_by_name.include?(name) && @project&.name != name
```

Since `@project` could be `nil`, it's probably safer to use the safe operator here.